packages/block-editor/src/components/block-list/block.js

```diff
@@ -99,10 +99,10 @@ function BlockListBlock( {
 			isSelected={ isSelected }
 			attributes={ attributes }
 			setAttributes={ setAttributes }
-			insertBlocksAfter={ isLocked ? undefined : onInsertBlocksAfter }
-			onReplace={ canRemove ? onReplace : undefined }
-			onRemove={ canRemove ? onRemove : undefined }
-			mergeBlocks={ canRemove ? onMerge : undefined }
+			insertBlocksAfter={ isLocked ? noop : onInsertBlocksAfter }
+			onReplace={ canRemove ? onReplace : noop }
```

Comment:
Using `noop` here causes the "should insert line breaks when using enter and shift-enter" e2e test failure.

I will try to look into the details tomorrow.